contracts/staking/contracts/src/staking_pools/MixinStakingPoolRewards.sol

```diff
@@ -118,32 +112,39 @@ contract MixinStakingPoolRewards is
     ///      withdrawing rewards and adding/removing dependencies on cumulative rewards.
     /// @param poolId Unique id of pool.
     /// @param member of the pool.
-    /// @param initialDelegatedStakeToPoolByOwner The member's delegated
-    ///        balance at the beginning of this transaction.
-    /// @param finalDelegatedStakeToPoolByOwner The member's delegated balance
-    ///        at the end of this transaction.
     function _withdrawAndSyncDelegatorRewards(
         bytes32 poolId,
-        address member,
-        IStructs.StoredBalance memory initialDelegatedStakeToPoolByOwner,
-        IStructs.StoredBalance memory finalDelegatedStakeToPoolByOwner
```

Comment:
I don't think these should be removed. As-per the function comment, these values are the delegated stake at the beginning and end of the transaction. 

Ex, before delegating and after delegating. Before withdrawing and after withdrawing. 

We're now just calling `loadUnsynced` and `loadSynced` which is not the same.

Reply:
That said, since this change has already been made, we might be able to make this work by removing an optimization.

Currently we're now passing in the `init` value of delegated stake, which is needed for most operations - like computing balances. The `final` value is now only needed when determining whether we depend on the CR for this epoch or not (in `_setCumulativeRewardDependenciesForDelegator`). We _could_ just blindly say we depend on this epoch. It'll create more state bloat, but it does simplify the code.

Reply:
> We're now just calling `loadUnsynced` and `loadSynced` which is not the same.

If you look at the codepaths where this function is called, it was _always_ using `loadUnsynced` and `loadSynced`.

Reply:
The only changes in functionality with this PR should be the bug fix - everything else is a refactor. If you see a spot where that is not the case let me know!

Reply:
@abandeali1 See [here](https://github.com/0xProject/0x-monorepo/blob/3.0/contracts/staking/contracts/src/stake/MixinStake.sol#L180). The unsynced is loaded before delegating and the synced is loaded after delegating.

Reply:
I see. I think we can revert this commit then: https://github.com/0xProject/0x-monorepo/pull/2207/commits/22a6de48ae3bc133f7d3f21ad506f7794127e8bb